torch/utils/data/datapipes/utils/common.py

```diff
@@ -27,6 +34,31 @@ def check_lambda_fn(fn):
         )
 
 
+def serialize_fn(fn, is_dill_available):
```

Comment:
`is_dill_available` doesn't need to be the argument. We can determine `is_dill_available` inside this function. Then, in terms of the API simplicity, users/developers don't need to provide such static variable for each function.

Besides, we can let this function handle all functions dynamically. We can provide a default `__getstate__`/`__setstate__` here https://github.com/pytorch/pytorch/blob/17b3ba148de5bca4da66b4045e451b245f916179/torch/utils/data/dataset.py#L317 and loop over `__dict__` to invoke this function.

Reply:
I updated to remove `is_dill_available` as an argument, and also created/updated the default `__getstate__`/`__setstate__` for `IterDataPipe` and `MapDataPipe`.